src/Moryx.AbstractionLayer/Products/ProductIdentity.cs

```diff
@@ -28,19 +29,43 @@ public ProductIdentity(string identifier, short revision)
         }
 
         /// <summary>
-        /// Create a product identity that represents the latest revision of a given identifier
+        /// Creates a product identity that represents the latest revision of a given identifier
         /// </summary>
         public static ProductIdentity AsLatestRevision(string identifier) => new ProductIdentity(identifier, LatestRevision);
 
         /// <summary>
         /// Create a product identity from a string. Counterpart to ToString() method
         /// </summary>
-        /// <param name="identifier"></param>
+        /// <param name="identityString">the ProductIdentity as string</param>
+        /// <returns></returns>
+        public static ProductIdentity Parse(string identityString)
+        {
+            var splitIdentity = identityString.Split('-');
+            if(splitIdentity.Length != 2 )
+                throw new FormatException("identityString should consist of <identity>-<revision> instead of "+identityString);
```

Comment:
I would also use the same implementation and `nameof` can be used for `identityString`.